jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ContextHandlerTest.java

```diff
@@ -891,4 +899,162 @@ void assertInContext(Context context, Request request)
                 assertThat(r, sameInstance(request));
         }
     }
+
+    @Test
+    public void testGraceful() throws Exception
```

Comment:
I would consider this test invalid, as I would expect a `GracefulHandler` to wrap the `ContextHandler` if we want to test gracefulness, no?

Reply:
Can be done either wrapped or inside.  I've updated the test to use GracefulHandler, so it is kind of a double test of that handler, but at least we test it with ContextHandler now.